app/Controllers/extensionController.php

```diff
@@ -42,7 +42,17 @@ public function indexAction() {
 	 */
 	protected function getAvailableExtensionList() {
 		$extensionListUrl = 'https://raw.githubusercontent.com/FreshRSS/Extensions/master/extensions.json';
-		$json = @file_get_contents($extensionListUrl);
+
+		$cacheFile = CACHE_PATH . '/extensions.json';
+		if (FreshRSS_Context::$user_conf->retrieve_extension_list === true) {
+			if (!file_exists($cacheFile) || (time() - filemtime($cacheFile) > 86400)) {
+				@file_put_contents($cacheFile, file_get_contents($extensionListUrl));
```

Comment:
Please consider using `httpGet()` from `lib_rss.php` in order to use cURL options such as proxy.
That function will need a new case for JSON regarding HTTP Accept header `application/json`.

Reply:
https://github.com/FreshRSS/FreshRSS/pull/6767